src/SGFTree.h

```diff
@@ -77,6 +78,8 @@ class SGFTree {
 
     bool m_initialized{false};
     KoState m_state;
+    bool m_loaded_timecontrol{false};
+    TimeControl m_timecontrol;
```

Comment:
Hmm, so this is a bit annoying. Most of the SGFTree objects only have a KoState m_state, which has no associated timecontrol.

The GameStates are all constructed in follow_mainline_state. So, we don't really need every SGFTree node to have a copy of this.

The SGF spec says:
```
        A node containing game-info properties is called a game-info node.
        There may be only one game-info node on any path within the tree,
        i.e. if some game-info properties occur in one node there may not be
        any further game-info properties in following nodes:
        a) on the path from the root node to this node.
        b) in the subtree below this node.
```

And time control things are game-info properties.

Maybe SGFTree could be modified to hold shared pointers to a SGFGameInfo object or something similar instead of copying them everywhere. But there's also plenty of game-info properties which are encoded into KoState anyway, so it wouldn't gain us much.

So maybe, for now, make SGFTree take a shared_ptr to a TimeControl object? That will eliminate m_loaded_timecontrol too.

Reply:
@gcp I think I've done what you proposed. Let me know if I misunderstood or got it wrong somehow. sgf loading still works :)